Unity/Tests/BananaClient/TcpAdapterTest.cs

```diff
@@ -11,13 +10,97 @@
 using NUnit.Framework;
 using UnityEngine.TestTools;
 
-public class TcpAdapterTest : MonoBehaviour
+
+public class MemoryStreamWrapper: Stream
+{
+    private long readPos, writePos;
+
+    private bool _Disposed {get; set;} = false;
+
+    private readonly MemoryStream ms;
```

Comment:
Just using byte[] as buffer may be easier.

Reply:
I will do a circular buffer later, memorystream is too much of a hassle

Reply:
SG